extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java

```diff
@@ -125,7 +128,7 @@ private void calculate() {
             if (matcher.find()) {
                 remoteAddress = parseFor(matcher.group(1).trim(), remoteAddress.port());
             }
-        } else if (forwardingProxyOptions.allowXForwarded) {
+        } else if (forwardingProxyOptions.allowXForwarded && trustedXForwardedProxyCheck.isProxyAllowed()) {
```

Comment:
Shouldn't this check also be for the forwarded header?

Reply:
Personally, I'd do it for forwarded only if there is interest from users.

Reply:
@michalvavrik It makes sense to have both types of headers treated the same way as far as this restriction is concerned, agree with Stuart; sorry for a delay

Reply:
Done.